### PR TITLE
Get connectivity config via DittoHeaders

### PR DESCRIPTION
**Pull request by stmaute**
Extend the ConnectivityConfigProvider interface with an additional method to retrieve the connectivity config via DittoHeaders.

### CODE REVIEW

File: services/connectivity/messaging/src/main/java/org/eclipse/ditto/services/connectivity/messaging/validation/HostValidator.java [hunk 1]
```diff
@@ -58,19 +58,19 @@
...
         this.blockedAddresses = calculateBlockedAddresses(blockedHostnames, loggingAdapter);
     }
 
     /**
      * Validate if connections to a host are allowed by checking (in this order):
      * <ul>
-     *     <li>blocklist is empty? this completely disables validation, every host is allowed</li>
-     *     <li>host is contained in allowlist? host is allowed</li>
+     *     <li>blocklist is empty this completely disables validation, every host is allowed</li>
+     *     <li>host is contained in allowlist host is allowed</li>
```
Comment (thjaeckle) [inline services/connectivity/messaging/src/main/java/org/eclipse/ditto/services/connectivity/messaging/validation/HostValidator.java:71]:
```suggestion
     *     <li>if the blocklist is empty, this completely disables validation, every host is allowed</li>
     *     <li>if the host is contained in the allowlist, the host is allowed</li>
```

File: services/connectivity/messaging/src/main/java/org/eclipse/ditto/services/connectivity/messaging/validation/HostValidator.java [hunk 2]
```diff
@@ -58,19 +58,19 @@
...
      *     <li>host is resolved to a blocked ip (loopback, site-local, multicast, wildcard ip)? host is blocked</li>
```
Comment (thjaeckle) [inline services/connectivity/messaging/src/main/java/org/eclipse/ditto/services/connectivity/messaging/validation/HostValidator.java:72]:
```suggestion
     *     <li>if the host is resolved to a blocked ip (loopback, site-local, multicast, wildcard ip), the host is blocked</li>
```

File: services/connectivity/messaging/src/main/java/org/eclipse/ditto/services/connectivity/messaging/validation/HostValidator.java [hunk 3]
```diff
@@ -58,19 +58,19 @@
...
-     *     <li>host is contained in the blocklist? host is blocked</li>
+     *     <li>host is contained in the blocklist host is blocked</li>
```
Comment (thjaeckle) [inline services/connectivity/messaging/src/main/java/org/eclipse/ditto/services/connectivity/messaging/validation/HostValidator.java:73]:
```suggestion
     *     <li>if the host is contained in the blocklist, the host is blocked</li>
```

**Review by thjaeckle — CHANGES_REQUESTED** [review]
I think the javadoc list should be rephrased - good that you removed the `?` in the comment, that was already confusing - I suggested to make complete sentences out of it.

File: services/connectivity/messaging/src/main/java/org/eclipse/ditto/services/connectivity/messaging/validation/HostValidator.java (hunk 1 above)
Comment (stmaute) [reply services/connectivity/messaging/src/main/java/org/eclipse/ditto/services/connectivity/messaging/validation/HostValidator.java:71]: It makes sense to extend the javadoc in this way 👍🏻

**Review by thjaeckle — APPROVED** [review]

**Pull request merged**